content/articles/2017-12-14-styler-1-0-0.Rmd

```diff
@@ -0,0 +1,230 @@
+
+```{r}
+library("styler")
+style_text("a=3; 2", scope = "spaces")
+```
+
+Or, on the other extreme of the scale, styling spaces, indention, line breaks 
+and tokens:
+
+```{r}
+style_text("a=3; 2", scope = "tokens")
+```
+
+In the next example, we want to style spaces and indention, but not line breaks
+and tokens.
+```{r}
+style_text(
+  "a=  function( x) {
+  'x'
+  }", 
+  scope = "indention"
+)
+```
+
+
+### `strict`
+
+Another option that is helpful to determine the level of 'invasiveness' is `strict`. If set to `TRUE`, spaces and line breaks before or after tokens are set to either zero or one. However, in some situations, this might be undesirable (so we set `strict = FALSE`), as the following example shows:
+
+```{r}
+style_text(
+  "data_frame(
+     small  = 2 ,
+     medium = 4,#comment without space
+     large  =6
+   )", 
+  strict = FALSE
+)
+```
+
+We prefer to keep the equal sign after "small", "medium" and large aligned, so we set `strict = FALSE` to set spacing to *at least* one around `=`.
+
+Also, with `scope >= "line_breaks"`, `strict = TRUE` will break the lines 
+in the following call, whreas `strict = FALSE` will leave them as is.
+```{r}
+style_text(
+  "do_a_long_and_complicated_fun_cal('which', has, way, to, 
+                              'and longer then lorem ipsum in its full length'
+                              )"
+)
+```
+
+
+In addition, the `strict` option does also adds braces (if `scope = "tokens"`) to conditional statements if they are multi-line.
+```{r}
+style_text(
+  "if (set_q(x))
+    3 else 
+    5",
+  strict = TRUE
+)
+```
+
+
+### `math_token_spacing`
+
+styler can identify and handle unary operators and other math tokens:
+
+```{styler}
+1++1-1-1/2
+```
+
+This is tidyverse style. However, styler offers very granular control for 
+math token spacing. Assuming you like spacing around `+` and `-`, but not 
+around `/` and `*` and `^`. This can be achieved as follows:
+```{r}
+style_text(
+  "1++1/2*2^2", 
+  math_token_spacing = specify_math_token_spacing(zero = c("'/'", "'*'", "'^'"))
+)
+```
+
+### `reindention`
+
+You can set the indention of lines that match a certain regular expression manually. For example, if you don't want comments starting with `###` to be indented, you can formulate an unindention rule:
+```{r}
+style_text(
+    "a <- function() {
+    ### not to be indented
+    # indent normally
+    33
+    }",
+  reindention = specify_reindention(regex_pattern = "^###", indention = 0)
+)
+```
+
+
+These are options that allow flexible reformatting. You can even go further, as described in [this](http://styler.r-lib.org/articles/customizing_styler.html) vignette.
+
+# A few more examples
+
+Now that you know that the rules can be applied in a flexible fashion, let's focus on `scope = "tokens"` and `strict = TRUE` and illustrate how specific rulres of the tidyverse style guide are implemented. 
```

Comment:
rulres

Reply:
Is it worth mentioning that all scopes below `"tokens"` are validated automatically, and that only `"tokens"` actually changes code?

Reply:
Yes.